Assignment 2.Rmd

```diff
@@ -95,6 +123,80 @@ Now practice with data from our class. Please create a **person-network** with t
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
+```{r}
+library(dplyr)
+library(tidyr)
+library(igraph)
+DFC<- read.csv("hudk4050-classes.txt")
+DF3<- select(DFC, Q8, Q9,Q1:Q7)
+DF3$name<-paste(DF3$Q8, DF3$Q9, sep = " ")
+DF3<- DF3[-c(36), ] #drop this person because s/he did not specify which class
+DF3<- DF3[-c(1,2),-c(1,2)]
+
+
+#Deal with classes
+
+DF4<- select(DF3, c(1:6))  
+DF4<- data.frame(lapply(DF4, as.matrix), stringsAsFactors = FALSE)
+
+#stack all the classes with names
+LQ1<- select(DF3, Q1,name)
+names(LQ1)<- c("class","name")
+LQ3<- select(DF3, Q3,name)
+names(LQ3)<- c("class","name")
+LQ4<- select(DF3, Q4,name)
+names(LQ4)<- c("class","name")
+LQ6<- select(DF3, Q6,name)
+names(LQ6)<- c("class","name")
+LQ7<- select(DF3, Q7,name)
+names(LQ7)<- c("class","name")
+LA<- rbind(LQ1,LQ3,LQ4,LQ6,LQ7)
+
+#remove empty col
+LA[LA==""] <- NA
+LA2<- na.omit(LA)
+#count
+LA3<-count(LA2, class, name)
+
+#remove empty space
+LA4<- apply(LA3,2,function(x)gsub('\\s+', '',x))
+#spread to matrix
+LA5<- as.data.frame(LA4, stringsAsFactors=FALSE)
+EdgeA <- spread(LA5, class, n, fill = 0)
+row.names(EdgeA)<- EdgeA$name
+EdgeA<- EdgeA[, -1]
+EdgeA<- as.matrix(EdgeA)
+
+#transpose matrix
+EdgeB<-as.matrix(t(EdgeA))
+#convert to numeric
+class(EdgeA) <- "numeric"
+class(EdgeB) <- "numeric"
+#combine two matrix
+EdgeC<- EdgeA %*% EdgeB
+diag(EdgeC)<- 0
+
+
+#graph
+g3 <- graph.adjacency(EdgeC,mode="undirected")
+deg <- centr_degree(g3, mode = "all")
+
+#Who is the most central person in the network?
+V(g3)$name[degree(g3)==max(degree(g3))]
+#"Artemas Wang" and "Lintong Li"
```

Comment:
Great job overall! Try to interpret the centrality measures within the context of this dataset.